middleware/github.js

```diff
@@ -56,11 +56,11 @@ async function setupUserClient(req, token) {
 
 // Get GitHub user info and attach it to the request
 async function setupInfo(req, cacheKey, client) {
-  let info = await req.app.locals.cache.get(cacheKey)
+  let info = await options.cache.get(cacheKey)
```

Comment:
Looking at this now I wonder if the cache could/should be passed in explicitly rather than as part of `options`. I don't recall why it is not. Just feels odd. Everything else in options is really just data, not functional.

Concretely I'm suggesting that the factory function at the end take an additional arg which is the cache service to use (or a factory or options to use to create the cache). Not sure this is a huge deal.  Just sticks out a bit.

Also, if we are using the injected option, then we don't need the middleware. The middleware just adds `req.app.local.cache` but all uses of that are now using the injected options object.